blocks.js

```diff
@@ -8744,6 +8847,13 @@ TemplateSlotMorph.prototype.evaluate = function () {
     return this.contents();
 };
 
+// TemplateSlotMorph to "scratchblocks"
+
+TemplateSlotMorph.prototype.stringify = function () {
+    var category = this.parent.stringifyCategory() || ' :: grey';
+    return '(' + this.children[0].stringify() + category + ')';
```

Comment:
Your demo site either isn't updated to the newest code, or your code is still buggy, because I'm still getting incorrect results for rings